Sources/TuistKit/Services/AuthTokenRefreshService.swift

```diff
@@ -0,0 +1,97 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+import TuistCore
+import TuistLoader
+import TuistServer
+import TuistSupport
+
+@Mockable
+protocol AuthTokenRefreshServicing: AnyObject {
+    func run(
+        directory: String?
+    ) async throws
+}
+
+final class AuthTokenRefreshService: AuthTokenRefreshServicing {
+    private let configLoader: ConfigLoading
+    private let fileSystem: FileSysteming
+    private let serverURLService: ServerURLServicing
+    private let refreshAuthTokenService: RefreshAuthTokenServicing
+    private let serverCredentialsStore: ServerCredentialsStoring
+    private let serverSessionController: ServerSessionControlling
+    private let serverAuthenticationController: ServerAuthenticationControlling
+
+    // MARK: - Init
+
+    init(
+        configLoader: ConfigLoading = ConfigLoader(warningController: WarningController.shared),
+        fileSystem: FileSysteming = FileSystem(),
+        serverURLService: ServerURLServicing = ServerURLService(),
+        refreshAuthTokenService: RefreshAuthTokenServicing = RefreshAuthTokenService(),
+        serverCredentialsStore: ServerCredentialsStoring = ServerCredentialsStore(),
+        serverSessionController: ServerSessionControlling = ServerSessionController(),
+        serverAuthenticationController: ServerAuthenticationControlling = ServerAuthenticationController()
+    ) {
+        self.configLoader = configLoader
+        self.fileSystem = fileSystem
+        self.serverURLService = serverURLService
+        self.refreshAuthTokenService = refreshAuthTokenService
+        self.serverCredentialsStore = serverCredentialsStore
+        self.serverSessionController = serverSessionController
+        self.serverAuthenticationController = serverAuthenticationController
+    }
+
+    func run(
+        directory: String?
+    ) async throws {
+        let directoryPath: AbsolutePath
+        if let directory {
+            directoryPath = try AbsolutePath(validating: directory, relativeTo: try await fileSystem.currentWorkingDirectory())
+        } else {
+            directoryPath = try await fileSystem.currentWorkingDirectory()
+        }
```

Comment:
This dance should be only necessary for "command services". This services should already have the necessary information passed through its parameters.